### PR TITLE
Update imgproxy to work with info endpoint

### PR DESCRIPTION
**Pull request by Awea**
This adds support for the [info](https://docs.imgproxy.net/usage/getting_info) endpoint

### CODE REVIEW

**Comment by bmuller** [comment]
Thanks for the contribution, @Awea!

The way the library currently works, it's possible to chain a number of options, like this:
```elixir
"https://placekitten.com/200/300"
|> Imgproxy.new()
|> Imgproxy.crop(100, 100)
|> Imgproxy.set_extension("jpg")
|> to_string()
```

I think by adding another public function that operates on an `Imgproxy` struct, the assumption given the existing pattern is that something like this could be possible:

```elixir
"https://placekitten.com/200/300"
|> Imgproxy.new()
|> Imgproxy.crop(100, 100)
|> Imgproxy.set_extension("jpg")
|> Imgproxy.info()
|> to_string()
```

Which I think would produce a URL (but should probably produce an error).

If I were going to add additional support for other endpoints, I think I'd probably pull the struct out of Imgproxy and put it in `Imgproxy.ProcessRequest`, and then do something like add another module/struct for `Imgproxy.InfoRequest` - and each module would have the functions necessary to add options.  The options helper functions might go in a `Imgproxy.Request` module.  So then the developer experience would be more like:

```elixir
alias Imgproxy.ProcessRequest

"https://placekitten.com/200/300"
|> ProcessRequest.new()
|> ProcessRequest.crop(100, 100)
|> ProcessRequest.set_extension("jpg")
|> to_string()
```

and

```elixir
alias Imgproxy.InfoRequest

"https://placekitten.com/200/300"
|> InfoRequest.new()
|> to_string()
```

WDYT?

**Comment by Awea** [comment]
I think it will be more clearer to use that way, I pushed some changes to reflect that, what do you think?

File: README.md [hunk 1]
```diff
@@ -64,6 +64,12 @@ Imgproxy.new("https://placekitten.com/200/300") |> to_string()
 |> Imgproxy.crop(100, 100)
 |> Imgproxy.set_extension("jpg")
 |> to_string()
+
+# Getting info for an image
+"https://placekitten.com/200/300"
+|> Imgproxy.new()
+|> Imgproxy.info()
+|> to_string()
```
Comment (bmuller) [inline README.md]: Think these examples may be outdated

Comment (Awea) [reply README.md]: Yes they where, it's fixed :pray:

File: mix.exs [hunk 2]
```diff
@@ -67,7 +67,8 @@ defmodule ImgProxy.MixProject do
   defp deps do
     [
       {:credo, "~> 1.5", only: [:dev, :test], runtime: false},
-      {:ex_doc, "~> 0.25", only: :dev, runtime: false}
+      {:ex_doc, "~> 0.25", only: :dev, runtime: false},
+      {:mix_test_watch, "~> 1.0", only: [:dev, :test], runtime: false}
```
Comment (bmuller) [inline mix.exs]: My preference would be to move any changes to dependencies to a different PR just to keep the PR's consistent in theme/purpose.

**Comment by Awea** [comment]
@bmuller Hello there! As we said here "Bon Bout d'An a l'an qué ven" :) Any other changes required for this PR to be merged and a new version released? :pray:

**Comment by bmuller** [comment]
Hey @Awea - after digging into this again, I have a few thoughts:
 1. I'm unable to test the info endpoint, since it's a paid feature and I don't have a license.  My ideal intention for this lib was/is to support the core URL generation capability for modifying images, this single new info endpoint is something I'd like to support but it's definitely a lower priority/concern
 2. I think the changes in this PR are pretty significant just to support 1 primary and 1 secondary URL endpoint (with the possibility of more endpoints pretty low, I think)
 3. I'd like to propose a much more limited modification that I think would still solve for your needs - see #6

Let me know what you think 🙏

**Comment by Awea** [comment]
Hey, I think it make sense to limit the changes for all the reason you said. I like the alternative you suggested (looks a bit like the first version of this PR :p), I'll update the PR that way.

**Comment by Awea** [comment]
Hey there, given a second thought I think we should go with #6 instead of this PR. I've tested it and everything works as intended.

I let you close this PR if you're okay with this :)

**Comment by bmuller** [comment]
Sounds good - I'll merge #6.  Thanks 🙏
